pkg/serializer/serializer.go

```diff
@@ -163,22 +165,46 @@ func (s Serializer) serializePayload(payload marshaler.Marshaler, compress bool,
 	return payloads, extraHeaders, nil
 }
 
-func (s Serializer) serializeStreamablePayload(payload marshaler.StreamJSONMarshaler) (forwarder.Payloads, http.Header, error) {
-	payloads, err := s.seriesPayloadBuilder.Build(payload)
+func (s Serializer) serializeStreamablePayload(payload marshaler.StreamJSONMarshaler, policy jsonstream.OnErrTooBigPolicy) (forwarder.Payloads, http.Header, error) {
+	payloads, err := s.seriesPayloadBuilder.BuildWithOnErrTooBigPolicy(payload, policy)
 	return payloads, jsonExtraHeadersWithCompression, err
 }
 
+func (s Serializer) serializeStreamJSONMarshalerFactoryPayload(
+	streamJSONMarshalerFactory marshaler.StreamJSONMarshalerFactory) (forwarder.Payloads, http.Header, error) {
+	marshaler := streamJSONMarshalerFactory.CreateSingleMarshaler()
+	eventPayloads, extraHeaders, err := s.serializeStreamablePayload(marshaler, jsonstream.FailedErrTooBig)
+
+	if err == jsonstream.ErrTooBig {
+		for _, v := range streamJSONMarshalerFactory.CreateMarshalerCollection() {
+			var eventPayloadsForSourceType forwarder.Payloads
```

Comment:
Let's add expvars to check if/where this happens. Also expvars for errors

Reply:
Added.